### PR TITLE
Fix issue with host header

### PR DESCRIPTION
**Pull request by Tuizi**
Add the possibility to set the `host` header.

### CODE REVIEW

**Comment by mcollina** [comment]
Good work! Can you update the docs and add a cli argument to set this as well?

**Comment by Tuizi** [comment]
There is no specific argument or doc needed I guess. It is just a "bug fix".
Before the `host` header was ignored no it is taken in consideration.

What do you think?

File: lib/httpRequestBuilder.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ function requestBuilder (defaults) {
     const headers = reqData.headers
     const body = reqData.body
 
-    let host = reqData.host
+    let host = reqData.headers.host || reqData.host
```
Comment (silverwind) [inline lib/httpRequestBuilder.js:40]: This does seem to fail when the header is given as `-H Host=value` on the CLI, e.g. it does not work case-insensitively like HTTP headers are supposed to be.

Comment (mcollina) [reply lib/httpRequestBuilder.js:40]: wow, this was a long time ago, would you open a fresh issue or a PR to fix?

Comment (silverwind) [reply lib/httpRequestBuilder.js:40]: https://github.com/mcollina/autocannon/pull/463

Comment (exoticknight) [reply lib/httpRequestBuilder.js:40]: I use
```bash
autocannon -H "host=xxx.com" "http://127.0.0.1:8000"
```
it seems that it just replace 127.0.0.1 with xxx.com in url which is not what i intend

```
Request {
  bodyUsed: false,
  headers: Headers { connection: "keep-alive", host: "xxx.com, xxx.com" },
  method: "GET",
  redirect: undefined,
  url: "http://xxx.com/"
}
```

Deno serve
```ts
Deno.serve((req) => {
  console.log(req)
  return new Response('a')
})
```

**Pull request merged**